mono/mini/mini-profiler.c

```diff
@@ -52,13 +52,17 @@ emit_fill_call_ctx (MonoCompile *cfg, MonoInst *method, MonoInst *ret)
 	MONO_EMIT_NEW_STORE_MEMBASE (cfg, OP_STORE_MEMBASE_REG, alloc->dreg, MONO_STRUCT_OFFSET (MonoProfilerCallContext, method), method->dreg);
 
 	if (ret) {
-		MonoInst *var = mono_compile_create_var (cfg, mono_method_signature_internal (cfg->method)->ret, OP_LOCAL);
+		MonoType *ret_type = mono_method_signature_internal (cfg->method)->ret;
+		MonoInst *var = mono_compile_create_var (cfg, ret_type, OP_LOCAL);
 
 		MonoInst *store, *addr;
 
 		EMIT_NEW_TEMPSTORE (cfg, store, var->inst_c0, ret);
 		EMIT_NEW_VARLOADA (cfg, addr, var, NULL);
 		MONO_EMIT_NEW_STORE_MEMBASE (cfg, OP_STORE_MEMBASE_REG, alloc->dreg, MONO_STRUCT_OFFSET (MonoProfilerCallContext, return_value), addr->dreg);
+
+		MonoInst *move_ret_back;
+		EMIT_NEW_VARSTORE (cfg, move_ret_back, ret, ret_type, var);
```

Comment:
Considering this instruction is there to deal with what I would think is really an issue in the register allocator, a comment explaining why it's needed is probably warranted. It's not obvious without the context of this PR.

Reply:
good point. also added a guard for that specific situation. although it terms of functionality it doesn't change anything, it makes the generated code a little worse.